src/util/LRUCache.h

```diff
@@ -163,8 +163,9 @@ class LRUCache {
   size_t _capacity;
   EntryList _data;
   AccessMap _accessMap;
-  // TODO(schnelle): Once we switch to C++17
-  // this should be a shared_mutex
+  // Sadly there is currently no easy way to upgrade a shared_lock to an
+  // exclusive lock. Thus using a shared_lock here would conflict with moving to
+  // the front of the queue in operator[]
```

Comment:
Is there any chance to change/ improve this sometime in the near future? In this case I would leave a TODO there to find it.